advanced/kernel-messaging/src/index.ts

```diff
@@ -28,31 +26,34 @@ namespace CommandIDs {
 const extension: JupyterFrontEndPlugin<void> = {
   id: 'kernel-messaging',
   autoStart: true,
-  requires: [ICommandPalette, ILauncher, IMainMenu],
+  optional: [ILauncher],
+  requires: [ICommandPalette, IMainMenu],
   activate: activate
 };
 
 function activate(
   app: JupyterFrontEnd,
   palette: ICommandPalette,
-  launcher: ILauncher,
-  mainMenu: IMainMenu
+  mainMenu: IMainMenu,
+  launcher: ILauncher | null
 ) {
   const manager = app.serviceManager;
   const { commands, shell } = app;
-  let category = 'Tutorial';
+  const category = 'Example';
 
   // Add launcher
-  launcher.add({
-    command: CommandIDs.create,
-    category: category
-  });
+  if (launcher) {
+    launcher.add({
+      command: CommandIDs.create,
+      category: category
+    });
+  }
 
   function createPanel() {
-    let panel: TutorialPanel;
+    let panel: ExamplePanel;
```

Comment:
Same question as for the other PR about using `async` / `await`.

Reply:
Yes this helps readability.